src/app/static/src/app/components/figures/graphs/layout.ts

```diff
@@ -0,0 +1,24 @@
+import {TransformationProps, useTransformation} from "../transformedData";
+import {Dictionary} from "vue-router/types/router";
+import {deepCopy} from "../../../utils";
+
+interface Props extends TransformationProps {
+    layout: Dictionary<any>
+}
+
+export function useLayout(props: Props) {
+    const layout = deepCopy(props.layout);
```

Comment:
Vaguely related to your point about extensions to Plotly - we're missing type definitions for [layout](https://github.com/mrc-ide/mintr/blob/master/inst/schema/Graph.schema.json#L74) etc. As a result this function is rather light on types...

Reply:
Good point - I'll try adding a bit more typing in my current ticket which also deals with layouts